lib/reek/context_builder.rb

```diff
@@ -101,93 +140,278 @@ def process_send(exp)
+    #
     def process_begin(exp)
       increase_statement_count_by(exp.children)
       decrease_statement_count
-      process_default(exp)
+      process(exp)
     end
 
     alias_method :process_kwbegin, :process_begin
 
+    # Handles `if` nodes.
+    #
+    # An input example that would trigger this method would be:
+    #
+    # if(a > 5 && b < 3)
+    #   puts 'bingo'
+    # else
+    #   3
+    # end
+    #
+    # Counts the `if` body as one statement and the `else` body as another statement.
+    #
+    # At the end we subtract one statement because the surrounding context was already counted
+    # as one (e.g. via `process_def`).
+    #
+    # `children[1]` refers to the `if` body (so `puts 'bingo'` from above) and
+    # `children[2]` to the `else` body (so `3` from above), which might be nil.
+    #
     def process_if(exp)
       children = exp.children
       increase_statement_count_by(children[1])
       increase_statement_count_by(children[2])
       decrease_statement_count
-      process_default(exp)
+      process(exp)
     end
 
+    # Handles `while` and `until` nodes.
+    #
+    # An input example that would trigger this method would be:
+    #
+    # while(x < 5)
+    #   puts 'bingo'
+    # end
+    #
+    # Counts the `while` body as one statement.
+    #
+    # At the end we subtract one statement because the surrounding context was already counted
+    # as one (e.g. via `process_def`).
+    #
+    # `children[1]` below refers to the `while` body (so `puts 'bingo'` from above)
+    #
     def process_while(exp)
       increase_statement_count_by(exp.children[1])
       decrease_statement_count
-      process_default(exp)
+      process(exp)
     end
 
     alias_method :process_until, :process_while
 
+    # Handles `for` nodes.
+    #
+    # An input example that would trigger this method would be:
+    #
+    # for i in [1,2,3,4]
+    #   puts i
+    # end
+    #
+    # Counts the `for` body as one statement.
+    #
+    # At the end we subtract one statement because the surrounding context was already counted
+    # as one (e.g. via `process_def`).
+    #
+    # `children[2]` below refers to the `while` body (so `puts i` from above)
+    #
     def process_for(exp)
       increase_statement_count_by(exp.children[2])
       decrease_statement_count
-      process_default(exp)
+      process(exp)
     end
 
+    # Handles `rescue` nodes.
+    #
+    # An input example that would trigger this method would be:
+    #
+    # def simple
+    #   begin
+    #     raise ArgumentError, 'raising...'
+    #   rescue => e
+    #   end
+    # end
+    #
+    # Counts the `rescue` body as one statement.
+    #
+    # At the end we subtract one statement because the surrounding context was already counted
+    # as one (e.g. via `process_def`).
+    #
+    # `exp.children.first` below refers to the everything that is covered by the `rescue`
+    # node body (which is everything from `begin` to `rescue` in the example above).
```

Comment:
Oh, maybe not, maybe that’s a `resbody` – I’d add an explicit note here about this, though.